File: Data Structures & Algorithms/validate-parentheses/submission-3.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> sk;
        for(char ch:s){
            switch(ch){
                case '{':
                case '(':
                case '[':
                    sk.push(ch);
                    break;
                case '}':
                    if(sk.empty() || sk.top()!='{')
                        return false;
                    sk.pop();
                    break;
                case ')':
                    if(sk.empty() || sk.top()!='(')
                        return false;
                    sk.pop();
                    break;
                case ']':
                    if(sk.empty() || sk.top()!='[')
                        return false;
                    sk.pop();
                    break;
            }
        }
        if(sk.empty())
            return true;
        return false;
    }
};

File: Data Structures & Algorithms/decode-string/submission-3.java
class Solution {
    public String decodeString(String s) {
        Stack<String> st2 = new Stack<>();
        String ans = "";
        String num = "";
        for(char ch: s.toCharArray()){
            if(ch >= '0' && ch <= '9'){
                num += ch;
            }
            if((ch >= 'a' && ch <= 'z') || (ch >= 'A' && ch <= 'Z')){
                st2.push(""+ch);
            }
            if(ch=='['){
                st2.push(num);
                st2.push(""+ch);
                num = "";
            }
            if(ch==']')
            {
                String temp = "";
                while(!st2.peek().equals("[")){
                    // System.out.println(st2.peek());
                    String c = st2.pop();
                    temp = c + temp;
                }
                st2.pop();
                int n = Integer.parseInt(st2.pop());
                for(int i=0;i<n;i++) {
                    st2.push(temp);
                }
            }
        }
        while(!st2.isEmpty()){
            ans = st2.pop() + ans;
        }
        return ans;
    }
}